### ISSUE TITLE
Name a global dataclip

### ISSUE THREAD

**Issue by taylordowns2000** [OpenFn/Lightning#311]
### User story

As an admin that wants to run a job with an arbitrary dataclip (#238) I want to be able to select that input dataclip from a list of saved and _named_ `globals`.

Consider the two types of globals:

**Type 1:** There are inputs I want to save to test my jobs. These, when saved, form a “test suite” of cases that we’d like to validate for a particular job in a flow. They are results from previous runs (save as a test input for job3 in the flow) and saved webhook requests.

**Type 2:** There are constants, or blobs of JSON data, that non-technical users want to build via a simple UI and, most commonly, store “mapping” data that links, e.g. a clinic code to it’s underlying clinic name.

### Details

Currently, there is a picklist which allows the user to change the type of a dataclip. This picklist should dissappear. Instead, they should have a button which says "save as global" and if they do so, they should be prompted to give that dataclip a name which follows the same naming pattern as projects (i.e., it's a slug.) and be able to edit the body of the dataclip.

Once it's saved as a global, they can **_not_** convert it back to a `run_result` or a `http_request`.

Once it's saved as a global, they **_can_** set the name to null, effectively removing it from the "named globals" list.

### Implementation notes

- [ ] add a nullable column on the dataclips table called `name`
- [ ] add this nullable name to the dataclip model
- [ ] disable the body editor when type is `run_result` or `http_request`
- [ ] remove the type picklist
- [ ] add a "Save as Global" button which, when clicked makes the `body` editable and displays the `name` field... from this altered view of the form the user may still "Cancel" or "Save".

### QA

- [ ] This has been QA’d by Amber, branch can be merged after review

**Comment by taylordowns2000** [comment]
needs spec for dataclip index

**Comment by stuartc** [comment]
@taylordowns2000 since we're resurrecting this issue, the description doesn't really slide into our current experience.

Could you elaborate on how you see this working with our current interface?

I've always imagined this as a dataclip that is named, so either you make one from scratch and name it at the same time or you take an existing one and give it a name (previously I've been keen on a star, for example "I use this dataclip a lot, I'd like to find it again by name and/or by favourited/starred status).

I'm not following why we'd need to remove the type checklist, that's a filter all dataclips, why wouldn't someone want that? Or is this an artefact of our old interface?

Since dataclips are tied to runs and their type determines how they get rendered out when a job is executed, I'm very much in favour of having every dataclip being immutable, so a saved/named dataclip doesn't just become read only; it becomes a new dataclip. 
This addresses concerns I have with a global dataclip getting used dozens of times while being editing over and over.

I think it's important we decide how much auditability we want to offer, we could have a "parent_id" for cloned (the verb I'm using for naming or editing a dataclip). So we have a true representation of what a job was run with.

So how does this sound as a high level update of the original spec?

- When creating a new (custom) dataclip, we want to be able to name it.
- When browsing existing dataclips while looking for something to run a job with, we'd like to be able to name/favourite them.

How does that sound? 

If that sounds good, then we need to answer the question: is this the only place we want to do this? The list of dataclips in the input pane (afaik) are only for the current job, therefore making the experience pretty difficult to manage things.

Also how would editing dataclips look?

Your thoughts here would be appreciated!

**Comment by midigofrank** [comment]
@taylordowns2000  Because these are "special" dataclips, they should not get deleted when deleting project history right?

**Comment by taylordowns2000** [comment]
Correct! History related to them should be deleted as usual, but not the named (i.e., "saved") clips themselves.
